lib/api-loader/page-object.js

```diff
@@ -48,7 +49,7 @@ class PageObjectLoader extends BaseCommandLoader {
       const parent = this.api.page;
       let namespace;
       if (Array.isArray(this.namespace) && this.namespace.length > 0) {
-        namespace = PageObjectLoader.unflatten(parent, this.namespace);
+        namespace = PageObjectLoader.unflatten(parent, this.namespace.slice());
```

Comment:
Unflatten actually modifies the passed in `namespace` variable. After `unflatten` the `this.namespace` variable only contains `[]`. Since this array was passed in via `setNamespace` "as value" (without `.slice()`) this will then affect all following nested page objects in the same folder. So it worked for 1 nested page object, but broke with multiple page objects...

Reply:
Nice catch!